### ISSUE TITLE
Mask MySQL engine credentials in `SHOW CREATE` commands

### ISSUE THREAD

**Issue by OmarBazaraa** [ClickHouse/ClickHouse#21623]
Whenever you create a MySQL engine table or database, you can still get the credentials in plain text.

```sql
CREATE DATABASE IF NOT EXISTS db_mysql
ENGINE = MySQL('mysql-host:port', 'database', 'user', 'password');
```

```sql
SHOW CREATE DATABASE db_mysql

Query id: efa65efe-fa1b-42cc-b931-f6acc21f9add

┌─statement──────────────────────────────────────────────────────────┐
│ CREATE DATABASE db_mysql                                           │
│ ENGINE = MySQL('mysql-host:port', 'database', 'user', 'password')  │
│ SETTINGS mysql_datatypes_support_level = ''                        │
└────────────────────────────────────────────────────────────────────┘
```

I'd like to get a masked version of the credentials, something like this:

```sql
ENGINE = MySQL('mysql-host:port', 'database', '***', '***')
```

Because whenever a user has read access on a database, he/she can issue a `SHOW CREATE` command and easily get the credentials.

**Comment by phantom943** [comment]
I know I'm a bit late to the party, but you can use [Named collections](https://clickhouse.com/docs/en/operations/named-collections/)

**Comment by OmarBazaraa** [comment]
Fixed in version 22.12 #43227
